packages/@webex/media-helpers/src/webrtc-core.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/no-misused-new */
+/* eslint-disable valid-jsdoc */
+/* eslint-disable require-jsdoc */
+import {
+  AudioDeviceConstraints,
+  createCameraTrack as wcmeCreateCameraTrack,
+  createDisplayTrack as wcmeCreateDisplayTrack,
+  createMicrophoneTrack as wcmeCreateMicrophoneTrack,
+  LocalDisplayTrack,
+  LocalMicrophoneTrack as WcmeLocalMicrophoneTrack,
+  LocalCameraTrack as WcmeLocalCameraTrack,
+  VideoDeviceConstraints,
+} from '@webex/internal-media-core';
+
+export {
+  LocalTrack,
+  LocalDisplayTrack,
+  LocalTrackEvents,
+  type TrackMuteEvent,
+} from '@webex/internal-media-core';
+
+export class LocalMicrophoneTrack extends WcmeLocalMicrophoneTrack {
+  private unmuteAllowed = true;
+
+  /**
+   * @internal
+   */
+  setUnmuteAllowed(allowed) {
+    this.unmuteAllowed = allowed;
+  }
+
+  /**
+   * @returns true if user is allowed to unmute the track, false otherwise
+   */
+  isUnmuteAllowed() {
+    return this.unmuteAllowed;
+  }
+
+  setMuted(muted: boolean): Promise<void> {
```

Comment:
webrtc-core's setMuted returns void. I've changed it here to return a promise, because it might fail if you're not allowed to unmute, so I think it's nicer that it rejects or resolves now, but the consequence is that many things become async now (see other changes in this PR like Meeting.preMedia() etc) so I'm not 100% sure if it's a good idea to return a promise here. The alternative is to return void, but throw if you're not allowed to unmute.

Reply:
I've decided to change this back to return void (not a promise) and just throw when unmuting is not allowed. This simplifies the code in many other places.